after/ftdetect/foam.vim

```diff
@@ -0,0 +1,13 @@
+" Last Change: 2021 Dec 13
+
+function! s:foamFile(path)
+    let l:lines = getline(1, 10)
+    for line in lines
+        if match(line, 'FoamFile') >= 0
+            set filetype=foam
+        endif
+    endfor
+endfunction
```

Comment:
not sure if we want to have such explicit fttype detects. I mean this will run for all users always while virtually nobody uses this format.

Reply:
Unfortunately, this inherited from the language itself, [this workspace](https://github.com/OpenFOAM/OpenFOAM-dev/tree/master/tutorials/basic/scalarTransportFoam/pitzDaily) shows some examples of files I'm supporting (but we're not limited to those files). Virtually, there is no way to tell if it is an OpenFOAM file unless we take a peek inside.

However, I see your point. Should I limit file type detection to certain filename patterns here, and instruct users of the language to put the content-aware auto-command themselves?

Btw thanks for taking the time to review!

Reply:
Is there a vim plugin that already handles this filetype? It would be specifically nice when READMEs or similar that talk about OpenFOAM files would not be highlighted as such

Reply:
There are a couple of plugins I know of, but they are outdated:
[vim-OpenFoam-syntax](https://github.com/effi/vim-OpenFoam-syntax)
[VimExtensionOpenfoam](https://bitbucket.org/shor-ty/vimextensionopenfoam/src)
[vim-foam](https://github.com/lervag/vim-foam)

and they all handle the file type detection in similar ways, the most feasible is the generic content-based one unfortunately. That's OK there, but not here. That's why I'm using path patterns now; a little better than "*".

Reply:
Can it happen that I accidentally mention the term `OpenFoam` in a normal C++ file when I'm in a OpenFoam project. Would then C++ still be highlighted properly through the injections?